sql/information_schema/information_schema.go

```diff
@@ -514,11 +517,102 @@ func collationsRowIter(ctx *Context, c *Catalog) RowIter {
 			c.PadSpace(),
 		})
 	}
-	return RowsToRowIter(rows...)
+	return RowsToRowIter(rows...), nil
 }
 
-func emptyRowIter(ctx *Context, c *Catalog) RowIter {
-	return RowsToRowIter()
+func triggersRowIter(ctx *Context, c *Catalog) (RowIter, error) {
+	var rows []Row
+	for _, db := range c.AllDatabases() {
+		triggerDb, ok := db.(TriggerDatabase)
+		if ok {
+			triggers, err := triggerDb.GetTriggers(ctx)
+			if err != nil {
+				return nil, err
+			}
+			var triggerPlans []*plan.CreateTrigger
+			for _, trigger := range triggers {
+				parsedTrigger, err := parse.Parse(ctx, trigger.CreateStatement)
+				if err != nil {
+					return nil, err
+				}
+				triggerPlan, ok := parsedTrigger.(*plan.CreateTrigger)
+				if !ok {
+					return nil, ErrTriggerCreateStatementInvalid.New(trigger.CreateStatement)
+				}
+				triggerPlans = append(triggerPlans, triggerPlan)
+			}
+
+			triggerPlans = analyzer.OrderTriggers(triggerPlans)
+			var beforeDelete []*plan.CreateTrigger
+			var beforeInsert []*plan.CreateTrigger
+			var beforeUpdate []*plan.CreateTrigger
+			var afterDelete []*plan.CreateTrigger
+			var afterInsert []*plan.CreateTrigger
+			var afterUpdate []*plan.CreateTrigger
+			for _, triggerPlan := range triggerPlans {
+				switch triggerPlan.TriggerTime {
+				case sqlparser.BeforeStr:
+					switch triggerPlan.TriggerEvent {
+					case sqlparser.DeleteStr:
+						beforeDelete = append(beforeDelete, triggerPlan)
+					case sqlparser.InsertStr:
+						beforeInsert = append(beforeInsert, triggerPlan)
+					case sqlparser.UpdateStr:
+						beforeUpdate = append(beforeUpdate, triggerPlan)
+					}
+				case sqlparser.AfterStr:
+					// OrderTriggers reverses AFTER, so we must reverse them again
```

Comment:
Probably better to refactor OrderTriggers so there's a version that doesn't do that, kinda confusing for it to be exported with this quirk